src/commands/createRepo/RepoCreateStep.ts

```diff
@@ -53,10 +54,20 @@ export class RepoCreateStep extends AzureWizardExecuteStep<IStaticWebAppWizardCo
             ext.outputChannel.appendLog(pushedBranch);
             progress.report({ message: pushedBranch });
 
-            wizardContext.branchData = (await client.repos.getBranch({ repo: newRepoName, owner: nonNullProp(wizardContext, 'orgData').login, branch: nonNullProp(branch, 'name') })).data;
+            // getBranch will return undefined sometimes, most likely a timing issue so try to retrieve it for a minute
+            const maxTimeout = Date.now() + 60 * 1000;
+            while (!wizardContext.branchData && Date.now() < maxTimeout) {
```

Comment:
I prefer to follow a pattern like this:
```typescript
while (true) {
    wizardContext.branchData = 
    if (got branchData or timed out) {
        break;
    } else {
        await delay
    }
}
```

Your current logic essentially has an extra `delay(2000)` after we already found the branch data, which is what I'm trying to avoid.

eslint doesn't like `while (true)`, but I think it's fine. Seems like we can disable the rule specifically for loops (as opposed to `if (true)`) which I might do.